ipa-macros/src/step/mod.rs

```diff
@@ -0,0 +1,32 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::parse_macro_input;
+
+/// Generate a state transition graph and the corresponding `StepNarrow` implementations for the
+/// IPA protocol.
+pub fn expand(_attr: TokenStream, input: TokenStream) -> TokenStream {
+    let item = parse_macro_input!(input as syn::Item);
+    let step_enum = match item {
+        syn::Item::Enum(e) => e,
+        _ => {
+            return syn::Error::new_spanned(item, "ipa_macros::step expects an enum")
+                .to_compile_error()
+                .into()
+        }
+    };
+
+    add_derives(&step_enum).into()
+}
+
+/// Add required derives for the step enum to work with the IPA protocol.
+fn add_derives(enum_item: &syn::ItemEnum) -> TokenStream {
+    let enum_ident = &enum_item.ident;
+    quote!(
+        #[derive(AsRefStr, Debug, Clone, Copy, PartialEq, Eq, Hash)]
+        #[strum(serialize_all = "snake_case")]
```

Comment:
I see why, but it is also coupling two concepts tightly together (step -> strum -> str). It will be problematic if we want our own custom `AsRef` for any reason (including but not limited to standards).

It may be worth splitting it off and just generating step stuff here

Reply:
Good point. I actually stopped using strum in the next diff because I need to support tuple enum variants. For this diff it should be okay as is.

Reply:
yea we can let people opt-out of strum if they want to, or use it by slapping `derive(AsRefStr)` if they really don't care about the strings